Source/WebInspectorUI/UserInterface/Views/MediaDetailsSidebarPanel.js

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+WI.MediaDetailsSidebarPanel = class MediaDetailsSidebarPanel extends WI.DOMDetailsSidebarPanel
+{
+    // Static
+    static StyleClassName = "media";
+
+    // Public
+    constructor(delegate)
+    {
+        class MediaDetailsSectionSimpleRow extends WI.DetailsSectionSimpleRow
+        {
+            set pendingValue(value)
+            {
+                this.#pendingValue = value;
+                this.#dirty = true;
+            }
+
+            updateValue()
+            {
+                if (!this.#dirty)
+                    return;
+                this.value = this.#pendingValue;
+                this.#pendingValue = null;
+                this.#dirty = false;
+            }
+
+            #pendingValue = null;
+            #dirty = null;
+        }
+
+        super("media-details", WI.UIString("Media"));
+
+        this.#ui.sourceRow = new MediaDetailsSectionSimpleRow(WI.UIString("Source", "Source @ Media Sidebar", "Title for Source row in Media Sidebar"));
```

Comment:
This would mean adding null-checks everywhere because I can't guarantee that initialLayout() has been called before attached(). This is (IMO) an example of a premature optimization. The cost of creating these rows is very low, and in addition to being the view, each row also stores its displayed value, so they pull double duty.

I did originally have all this in initialLayout(), and moved them into the constructor for this very reason.

Reply:
perhaps in this case it might be premature because these objects are small, but it is the general practice of Web Inspector to not initialize view related stuff inside the `constructor` instead of `initialLayout` precisely because we've had issues with that in the past (e.g. i spent an entire summer moving things around because the aggregate of many cases like this got so bad)

while null checks would be fine you also could use `this.didInitialLayout` as is done in many other places.  in fact i think you would only need to wrap the call to `this.#startUpdateTimer();` inside `attached()` since that's really the only entrypoint that isn't during/after `initialLayout()`

Reply:
If there's a problem with creating all the panels before they're needed, then that needs to be resolved by... not creating all the panels until they're needed, and not by micro-optimizing the constructors of each individual panel.